lib/git_worktree.rb

```diff
@@ -258,6 +258,22 @@ def checkout
     end
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  #
+  def check_connection(url)
+    require 'net/ping/http'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
+    url = url.sub(':', '/').sub('git@', 'https://www.') if url.start_with?('git@')
```

Comment:
I don't really love this, but I can't think of anything else.  This seems like a decent compromise until we can do the built-in check_connection.  At a minimum it can verify connectivity.

However it assumes there is an https endpoint available, which technically it not always true.  From a git perspective, if all we are checking is connectivity, then a ping to the hostname without trying to ping a literal http endpoint should be sufficient.

Another option might be to use net/ssh (or perhaps manageiq-ssh_util) and try to ssh into the endpoint.  This would at least attempt to also verify things like the SSH key which is missing here.  With GitHub at least if you try to SSH in, it essentially verifies your SSH key...

```text
$ ssh git@github.com
PTY allocation request failed on channel 0
Hi Fryguy! You've successfully authenticated, but GitHub does not provide shell access.
Connection to github.com closed
```

Reply:
Seems like a different way to skin the same cat, as we'd then have to parse out the hostname from whatever they provided as their repo, and hope that sshd is enabled.

Reply:
Parsing out the hostname is simple... more or less `URL(url).hostname`

> I think in terms of how this will actually be used in practice, what I have here should be fine.

Agreed that SSH might be overkill but my comment about pinging a hostname vs an http endpoint still stands, I think.

Reply:
Ok, updated to just use an external ping on the hostname.